source/s3_request_messages.c

```diff
@@ -394,8 +421,19 @@ struct aws_http_message *aws_s3_message_util_copy_http_message(
             goto error_clean_up;
         }
 
+        if (multipart_upload_ops) {
+            if (aws_byte_cursor_eq_c_str_ignore_case(&header.name, "host") ||
+                aws_byte_cursor_eq_c_str_ignore_case(&header.name, "x-amz-request-payer") ||
+                aws_byte_cursor_eq_c_str_ignore_case(&header.name, "x-amz-expected-bucket-owner")) {
+                if (aws_http_message_add_header(message, header)) {
+                    goto error_clean_up;
+                }
+            }
+            continue;
+        }
+
         /* For SSE upload, the sse related headers should only be shown in the create_multipart_upload.*/
-        if (aws_byte_cursor_eq_c_str_ignore_case(&header.name, "x-amz-server-side-encryption") && !sse) {
+        if (aws_byte_cursor_eq_c_str_ignore_case(&header.name, "x-amz-server-side-encryption") && !sse_included) {
```

Comment:
Trivial: do the !sse_included check first since it's faster.

Reply:
yeah, that's fixed by the following pr https://github.com/awslabs/aws-c-s3/pull/53#discussion_r555345375